src/account/default.ts

```diff
@@ -573,23 +574,59 @@ export class Account extends Provider implements AccountInterface {
     return getMessageHash(typedData, this.address);
   }
 
-  public async verifyMessageHash(hash: BigNumberish, signature: Signature): Promise<boolean> {
+  public async verifyMessageHash(
+    hash: BigNumberish,
+    signature: Signature
+  ): Promise<SignatureVerifResult> {
     try {
-      await this.callContract({
+      const resp = await this.callContract({
         contractAddress: this.address,
         entrypoint: 'isValidSignature',
         calldata: CallData.compile({
           hash: toBigInt(hash).toString(),
           signature: formatSignature(signature),
         }),
       });
-      return true;
-    } catch {
-      return false;
+      // console.log('verifySign=', resp);
+      if (BigInt(resp.result[0]) === 0n) {
+        // OpenZeppelin 0.8.0 invalid signature
+        return {
+          isVerificationProcessed: true,
+          isSignatureValid: false,
+        } as SignatureVerifResult;
+      }
+      // OpenZeppelin 0.8.0, ArgentX 0.3.0 & Braavos Cairo 0 valid signature
+      return {
+        isVerificationProcessed: true,
+        isSignatureValid: true,
+      } as SignatureVerifResult;
+    } catch (err) {
+      // console.log('verifySign error=', err);
+      if ((err as Error).message.includes('argent/invalid-signature')) {
+        // ArgentX 0.3.0 invalid signature
+        return {
+          isVerificationProcessed: true,
+          isSignatureValid: false,
+        } as SignatureVerifResult;
+      }
+      if ((err as Error).message.includes('is invalid, with respect to the public key')) {
+        // Braavos Cairo 0 invalid signature
+        return {
+          isVerificationProcessed: true,
+          isSignatureValid: false,
+        } as SignatureVerifResult;
+      }
+      return {
+        isVerificationProcessed: false,
+        error: new Error('Signature verification request is rejected by the network.'),
+      } as SignatureVerifResult;
```

Comment:
isVerificationProcessed seems redundant with an error?
We should include err in Error?

I am more aligned with throwing errors:
1. This could silently break someone's code by not breaking after the update.
2. If the user does not want to break on network issue it can try-catch, as a user of networking libs I would assume it will throw on network issue.

Reply:
Throwing Error implemented.